tests/testautotls.nim

```diff
@@ -0,0 +1,45 @@
+{.used.}
+
+# Nim-Libp2p
+# Copyright (c) 2023 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+{.push raises: [].}
+
+# import std/[strformat, net] # uncomment after re-enabling AutoTLSManager
+import chronos
+import chronos/apps/http/httpclient
+import ../libp2p/[stream/connection, upgrademngrs/upgrade, autotls/acme/api, wire]
+
+import ./helpers
+
+suite "AutoTLS":
+  teardown:
+    checkTrackers()
+
+  asyncTest "test ACME":
```

Comment:
consider adding more tests, ones where error happen.
- register with unsupported key
- requestChallenge with unsupported key + invalid kid

consider testing other endpoints:
- finalizeCertificate
- downloadCertificate
- notifyChallengeCompleted

Reply:
testing other endpoints requires the autotls broker e.g. you can't finalize a certificate that has not passed the challenge test (passing the challenge test requires autotls broker)

Reply:
got it! we can still test and assert that methods are raising exceptions.